checks/apps/vasp/vasp_check.py

```diff
@@ -1,124 +1,179 @@
+        self.env_vars['OMP_NUM_THREADS'] = self.num_cpus_per_task
+
+        if self.uarch == 'gh200':
+            self.env_vars['MPICH_GPU_SUPPORT_ENABLED'] = '1'
+            self.env_vars['NCCL_IGNORE_CPU_AFFINITY'] = '1'
+
+        # set reference
+        if self.uarch is not None and \
+           self.uarch in vasp_references[self.test_name]:
+            self.reference = {
+                self.current_partition.fullname:
+                    vasp_references[self.test_name][self.uarch][self.num_nodes]
             }
-        }
-    }
 
+
+    @sanity_function
+    def assert_reference(self):
+        force = sn.extractsingle(r'1 F=\s+(?P<result>\S+)',
+                                 self.stdout, 'result', float)
+        return sn.assert_reference(force, self.force_reference, -1e-5, 1e-5)
+
+    # INFO: The name of this function needs to match with the reference dict!
     @performance_function('s')
     def elapsed_time(self):
         return sn.extractsingle(r'Elapsed time \(sec\):'
                                 r'\s+(?P<time>\S+)', 'OUTCAR',
                                 'time', float)
 
+
+@rfm.simple_test
+class VaspBuildTest(rfm.CompileOnlyRegressionTest):
+    '''
+    Test VASP build from source.
+    '''
+
+    descr = 'VASP Build Test'
+    version = variable(str, value='6.4.3')
+    valid_prog_environs = ['+vasp-dev']
+    valid_systems = ['*']
+    build_system = 'Make'
+    # only build std target
+    maintainers = ['SSA']
+    # run on node to load uenv
+    build_locally = False
+    tags = {'uenv'}
+
+    @run_before('compile')
+    def prepare_build(self):
+        self.build_system.options = ['DEPS=1', 'std']
+
+        self.skip_if_no_procinfo()
+        cpu = self.current_partition.processor
+        self.build_system.max_concurrency = cpu.info['num_cpus_per_socket']
+
+        # Don't set FC variable, which breaks the makefile
+        self.build_system.flags_from_environ = False
+
+        self.uarch = uenv.uarch(self.current_partition)
+        self.build_system.builddir = self.stagedir
+        self.build_time_limit = '0d1h0m0s'
+
+        makefile = f'makefile.include.{self.uarch}'
+        makefile_path = os.path.join(self.prefix, self.sourcesdir, makefile)
+
+        if not os.path.isfile(makefile_path):
+            self.skip(f'No makefile for uarch {self.uarch}')
+
+
+        vasp_download_cmd = f'curl --retry 5 -u ${{CSCS_REGISTRY_USERNAME}}:${{CSCS_REGISTRY_PASSWORD}} -X GET https://jfrog.svc.cscs.ch/artifactory/uenv-sources/vasp/vasp-{self.version}.tar.bz2 -o vasp_src.tar.bz2'
+
+        # Download source and copy makfile matching uarch
+        self.prebuild_cmds = [
+            vasp_download_cmd,
+            'tar -xf vasp_src.tar.bz2',
+            # The vasp tar ball contains inconsistent directory names between versions,
+            # so we find the directory name and change it to vasp_src
+            'find . -maxdepth 1 -type d -name "vasp*" -exec mv {} vasp_src \;',
```

Comment:
```suggestion
            'find . -maxdepth 1 -type d -name "vasp*" -exec mv {} vasp_src \\;',
```
I get `W605 invalid escape sequence '\;'` with pycodestyle.